eventuals/catch.h

```diff
@@ -42,6 +59,85 @@ struct _Catch {
       e->Start();
     }
 
+    template <
+        typename Error_,
+        typename CatchHandler_,
+        typename... Tail>
+    void UnpackInvokeHelper(
+        Error_&& error,
+        CatchHandler_&& handler,
+        Tail&&... tail) {
+      if constexpr (
+          IsSame<
+              std::decay_t<CatchHandler_>,
+              std::decay_t<Error_>>::value) {
```

Comment:
Let's do something like:
```suggestion
      if constexpr (Handler::Handles<Error>) {
```